perf/src/main/java/org/bf2/performance/InstanceProfiler.java

```diff
@@ -424,14 +424,13 @@ private void setup() throws Exception {
         }
     }
 
-    private AvailableResources getMinAvailableResources(Stream<Node> stream) {
+    private AvailableResources getMinAvailableResources(Stream<Node> stream) throws NullPointerException{
         AvailableResources resources = stream.map(TestUtils::getMaxAvailableResources)
                 .reduce((a1, a2) -> {
                     a1.cpuMillis = Math.min(a1.cpuMillis, a2.cpuMillis);
                     a1.memoryBytes = Math.min(a1.memoryBytes, a2.memoryBytes);
                     return a1;
-                })
-                .get();
+                }).orElse(null);
```

Comment:
I don't think this is the right approach when the stream is empty. @shawkins , do you have any thoughts for a sensible identity value to pass to `reduce` to avoid the `Optional` here?

Reply:
@MikeEdgar I'm ok with just allowing the call to .get().  Forcing an alternative to the implicit NoSuchElementException will likely lead to adding a utility method that does exactly what you have as a replacement for get.

Reply:
Yes, good point. @biswassri this can remain unchanged, but use `//NOSONAR` to silence the warning.